tests/js/realm-tests.js

```diff
@@ -1220,6 +1220,51 @@ module.exports = {
         }, 'The Realm file format must be allowed to be upgraded in order to proceed.');
     },
 
+    testSchemaUpdates: function() {
+        return new Promise((resolve, reject) => {
+            let calls = 0;
+            let realm1 = new Realm({_cache: false});
+            TestCase.assertEqual(realm1.schema.length, 0);  // empty schema
+            realm1.addListener('schema', (realm, event, schema) => {
+                calls++;
+
+                TestCase.assertEqual(event, 'schema');
+                TestCase.assertEqual(schema.length, 1);
+                TestCase.assertEqual(realm.schema.length, 1);
+                TestCase.assertEqual(schema[0].name, 'TestObject');
+                TestCase.assertEqual(realm1.schema.length, 1);
+                TestCase.assertEqual(realm.schema[0].name, 'TestObject');
+
+                if (calls == 2) {
+                    resolve();
+                }
+            });
+
+            realm1.addListener('change', (realm, event) => {
+                calls++;
```

Comment:
This is minor, but the way this test is structured, if for some reason we get two notifications for changes, this will pass the test, even though we didn't get any schema notifications. I would suggest using a hash (or an object) to store the types of event notifications we've received and only resolve when we've received both `change` and `schema`.